server/service/global_policies.go

```diff
@@ -155,20 +155,53 @@ func deleteGlobalPoliciesEndpoint(ctx context.Context, request interface{}, svc
 	return deleteGlobalPoliciesResponse{Deleted: resp}, nil
 }
 
+// DeleteGlobalPolicies deletes the given policies from the database.
+// It also deletes the given ids from the failing policies webhook configuration.
 func (svc Service) DeleteGlobalPolicies(ctx context.Context, ids []uint) ([]uint, error) {
 	if err := svc.authz.Authorize(ctx, &fleet.Policy{}, fleet.ActionWrite); err != nil {
 		return nil, err
 	}
 	if len(ids) == 0 {
 		return nil, nil
 	}
+	if err := svc.removeGlobalPoliciesFromWebhookConfig(ctx, ids); err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "removing global policies from webhook config")
+	}
 	ids, err := svc.ds.DeleteGlobalPolicies(ctx, ids)
 	if err != nil {
-		return nil, err
+		return nil, ctxerr.Wrap(ctx, err, "deleting global policies")
 	}
 	return ids, nil
 }
 
+func (svc Service) removeGlobalPoliciesFromWebhookConfig(ctx context.Context, ids []uint) error {
+	ac, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return err
+	}
+	idSet := make(map[uint]struct{})
+	for _, id := range ids {
+		idSet[id] = struct{}{}
+	}
+	n := 0
+	policyIDs := ac.WebhookSettings.FailingPoliciesWebhook.PolicyIDs
+	origLen := len(policyIDs)
+	for i := range policyIDs {
+		if _, ok := idSet[policyIDs[i]]; !ok {
+			policyIDs[n] = policyIDs[i]
+			n++
+		}
+	}
+	if n == origLen {
+		return nil
+	}
+	ac.WebhookSettings.FailingPoliciesWebhook.PolicyIDs = policyIDs[:n]
```

Comment:
So this is for changing in place instead of generating another slice of ids, right? It feels like a premature optimization. Maybe if we had 100k+ policies here, but I can't imagine this being more than a few hundred.

Reply:
Ya, prolly is, I usually use this approach from (filter in place) https://ueokande.github.io/go-slice-tricks/.